### ISSUE TITLE
`step_select` fails when tidyselection selects case weights variable

### ISSUE THREAD

**Issue by TylerGrantSmith** [tidymodels/recipes#970]
Using `step_select` with a tidyselection operator that will include the case weights variable causes an error.

``` r
library(tidymodels)
set.seed(1)
x <- 
  sim_classification(5000) %>% 
  mutate(case_weights = importance_weights(1))

xgb_spec <- 
  boost_tree() %>% 
  set_mode("classification") %>% 
  set_engine("xgboost")

sim_rec <- 
  recipe(class ~ ., data = x) %>% 
  step_select(everything()) %>%
  step_normalize(all_numeric_predictors())

xgb_wflow <- 
  workflow() %>% 
  add_recipe(sim_rec) %>% 
  add_model(xgb_spec) 

xgb_res <- xgb_wflow %>% fit(data = x)
#> Error in `recipes_eval_select()`:
#> ! Cannot select case weights variable.
```

**Comment by EmilHvitfeldt** [comment]
Individual steps are not allowed to interact with the case weights columns, see `?recipes:::case_weights` for more information (or report back if the documentation isn't adequate).

As far as I can tell you are selecting the case weights columns because you are using `everything()`. This would break during prediction anyways since it would try to select the outcome. Can you describe what you were trying to do when you selected the case weights column?

We are aware that `step_select()` can be hard to use correctly in combination with workflows https://github.com/tidymodels/recipes/issues/741.

**Comment by TylerGrantSmith** [comment]
In my specific scenario I was using it to exclude some variables from the data using `-one_of(...)`  I can just as well do that to the data beforehand but I figured that the step exists with `recipes` for this sort of reason.  I think then that there is nothing new to this issue that isn't addressed in #741  
Thanks!

**Comment by EmilHvitfeldt** [comment]
If you want to exclude variables, you will be better off using `step_rm()`, this way it won't give you issues with the outcome or case weights.
